### PR TITLE
SCMOD-7821: Tomcat base image logging with logback

### PR DESCRIPTION
**Pull request by thilagasanthosh**
https://portal.digitalsafe.net/browse/SCMOD-7821

### CODE REVIEW

File: caf-logging-logback-tomcat/src/main/java/com/github/cafapi/logging/logback/tomcat/CAFAccessLogConsoleAppender.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        logMsgBuilder.append(ProcessAndThreadIdProvider.getId());
+        logMsgBuilder.append(" ");
+        if(event.getStatusCode() >= 400 && event.getStatusCode() <=499){
+            logMsgBuilder.append("WARN ");
+        }else if(event.getStatusCode() >= 500 && event.getStatusCode() <=599){
+            logMsgBuilder.append("ERROR");
+        }else{
+            logMsgBuilder.append("INFO ");
+        }
+        event.setThreadName(logMsgBuilder.toString());
```
Comment (andyreidz) [inline caf-logging-logback-tomcat/src/main/java/com/github/cafapi/logging/logback/tomcat/CAFAccessLogConsoleAppender.java]: You're highjacking the threadname for the log level, instead investigate a custom layout implementation

**Review by andyreidz — CHANGES_REQUESTED** [review]

File: caf-logging-logback-tomcat/src/main/java/com/github/cafapi/logging/logback/tomcat/CAFAccessLogConsoleAppender.java (hunk 1 above)
Comment (thilagasanthosh) [reply caf-logging-logback-tomcat/src/main/java/com/github/cafapi/logging/logback/tomcat/CAFAccessLogConsoleAppender.java]: A new converter added for Loglevel and a new encoder created to use the new converter.

**Comment by thilagasanthosh** [comment]
I am trying to use CAFConsoleAppender or new Encoder for processId and thread id in logback.xml for tomcat.
But the external classes are not loaded in logback.xml.
As per the soluiton mentioned in the issue https://github.com/tomcat-slf4j-logback/tomcat-slf4j-logback/issues/260 , the third party jars can be combined with tomcat-juli jar to use it for Custom encoder or Appender.
I am trying this currently.

**Comment by thilagasanthosh** [comment]
@dermot-hardy ,

I have referred the pom.xml for tomcat9-slf4j-logback and created new Uber jar project caf-logging-tomcat-juli.
https://github.com/tomcat-slf4j-logback/tomcat9-slf4j-logback/blob/master/pom.xml
The tomcat is starting fine and no issues with Documentid service project.

http://sou-jenkins2.swinfra.net/job/caf/job/caf~documentid-service~SCMOD-7821~CI/14/console
But I would like to get your comments as well for any extra entries or any entries should be removed from the new uber jar.

File: caf-logging-tomcat-juli/src/main/java/com/github/cafapi/logging/tomcat/juli/CAFTomcatConsoleAppender.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ * limitations under the License.
+ */
+package com.github.cafapi.logging.tomcat.juli;
+
+import org.apache.juli.logging.ch.qos.logback.classic.spi.LoggingEvent;
+import org.apache.juli.logging.ch.qos.logback.core.ConsoleAppender;
+
+import com.github.cafapi.logging.common.ProcessAndThreadIdProvider;
+
+public final class CAFTomcatConsoleAppender extends ConsoleAppender<LoggingEvent>
```
Comment (dermot-hardy) [inline caf-logging-tomcat-juli/src/main/java/com/github/cafapi/logging/tomcat/juli/CAFTomcatConsoleAppender.java:23]: What is the purpose of this class?  I can't find any references to it anywhere.  Also it just looks the same as `CAFConsoleAppender` so why can't you use that?

File: caf-logging-tomcat-juli/pom.xml [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+                                    <resource>META-INF/LICENSE-logback.txt</resource>
+                                    <file>${project.build.directory}/tomcat-slfj4-logback-config/assembly/LICENSE-logback.txt</file>
+                                </transformer>
+                                <transformer implementation="org.apache.maven.plugins.shade.resource.IncludeResourceTransformer">
+                                    <resource>META-INF/LICENSE-slf4j.txt</resource>
+                                    <file>${project.build.directory}/tomcat-slfj4-logback-config/assembly/LICENSE-slf4j.txt</file>
+                                </transformer>
+                                <transformer implementation="org.apache.maven.plugins.shade.resource.IncludeResourceTransformer">
+                                    <resource>META-INF/LICENSE-tomcat.txt</resource>
+                                    <file>${project.build.directory}/tomcat-slfj4-logback-config/assembly/LICENSE-tomcat.txt</file>
```
Comment (dermot-hardy) [inline caf-logging-tomcat-juli/pom.xml:207]: You seem to have these files checked in but not in this location, so I suspect that all these references are wrong.

File: caf-logging-tomcat-juli/pom.xml [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+                                <relocation>
+                                    <pattern>logback.configurationFile</pattern>
+                                    <shadedPattern>juli-logback.configurationFile</shadedPattern>
+                                </relocation>
+                                <!-- Located in org.apache.juli.logging.org.slf4j.MarkerFactory -->
+                                <relocation>
+                                    <pattern>logback.ContextSelector</pattern>
+                                    <shadedPattern>juli-logback.ContextSelector</shadedPattern>
+                                </relocation>
+                            </relocations>
```
Comment (dermot-hardy) [inline caf-logging-tomcat-juli/pom.xml:189]: How have you verified that all these filters and relocations are correct?

**Review by dermot-hardy — CHANGES_REQUESTED** [review]
This whole module shouldn't be in this repository.  You either need to make [CAFapi/opensuse-tomcat-image](https://github.com/CAFapi/opensuse-tomcat-image) into a multi-module project or else create a whole new repository for it.

**Comment by dermot-hardy** [comment]
I've closed the PR because the project needs to be moved.

**Pull request closed without merging**